pkg/lock/file.go

```diff
@@ -64,6 +64,10 @@ func TryExclusiveLock(path string, lockType LockType) (*FileLock, error) {
 	if err != nil {
 		return nil, err
 	}
+	if err = verifySameFile(l, path); err != nil {
```

Comment:
I don't think this check really adds anything in the `Try` variants of the functions.